mijia/src/bluetooth/bluetooth_event.rs

```diff
@@ -1,146 +0,0 @@
-// This file is copied from:
-//     https://github.com/szeged/blurz/blob/master/src/bluetooth_event.rs
-//
-// Copyright (c) 2016, University of Szeged
-// Copyright (c) 2016, Attila Dusnoki <adusnoki@inf.u-szeged.hu>
-// All rights reserved.
-//
-// Redistribution and use in source and binary forms, with or without
-// modification, are permitted provided that the following conditions are met:
-//
-// * Redistributions of source code must retain the above copyright notice, this
-//   list of conditions and the following disclaimer.
-//
-// * Redistributions in binary form must reproduce the above copyright notice,
-//   this list of conditions and the following disclaimer in the documentation
-//   and/or other materials provided with the distribution.
-//
-// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
-// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
-// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
-// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
-// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
-// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-use dbus::{arg::cast, arg::RefArg, arg::TypeMismatchError, arg::Variant, Message};
-use std::collections::HashMap;
-
-use crate::bluetooth::{AdapterId, CharacteristicId, DeviceId};
-
-#[derive(Clone, Debug)]
-pub enum BluetoothEvent {
-    Powered {
-        adapter: AdapterId,
-        powered: bool,
-    },
-    Discovering {
-        adapter: AdapterId,
-        discovering: bool,
-    },
-    Connected {
-        device: DeviceId,
-        connected: bool,
-    },
-    ServicesResolved {
-        device: DeviceId,
-        services_resolved: bool,
```

Comment:
Would ServicesResolved allow us to solve that bug when you connect() and then start_notify() and bluez says that it can't find the reading property's subscribe method and bombs out?

Reply:
Possibly. But maybe that should be something internal to the library? It needs some more thought, in any case, and I don't think there's much value in exposing the event to client until we have a story for how it should be used.